app/models/user.rb

```diff
@@ -175,8 +175,18 @@ class User < ApplicationRecord
 
   validate :profile_picture_format
 
+  def self.with_2fa_requirement_disabled(&)
+    previous_value = Fiber[:user_2fa_requirement_disabled]
+    Fiber[:user_2fa_requirement_disabled] = true
+    yield
+  ensure
+    Fiber[:user_2fa_requirement_disabled] = previous_value
+  end
+
   validate on: :update do
-    if Rails.env.production? && admin_override_pretend? && !use_two_factor_authentication?
+    next if Fiber[:user_2fa_requirement_disabled] || Rails.env.development?
+
+    if admin_override_pretend? && !use_two_factor_authentication?
```

Comment:
I'm not super comfortable with this - I don't like selectively running validations. The reason we have this validation on user is so that users can't disable their 2FA requirement.

Reply:
> I'm not super comfortable with this - I don't like selectively running validations.

Same. This is a duct tape solution that tries to solve the problem with minimal changes.

> The reason we have this validation on user is so that users can't disable their 2FA requirement.

Then I'd propose we do two things:

1. Keep the validation but only apply it if `use_two_factor_authentication` has changed, which will prevent users from disabling it and won't hinder other code paths that don't touch that attribute
2. Add a `before_filter` somewhere to force admins who don't have 2fa enabled to do so before doing anything else

Reply:
I'll update this PR to do (1) as it would effectively solve the bug, and I'll follow up with (2)